derp/derp_client.go

```diff
@@ -457,9 +461,8 @@ func (c *Client) recvTimeout(timeout time.Duration) (m ReceivedMessage, err erro
 			c.readErr.Store(err)
 		}
 	}()
-
 	for {
-		c.nc.SetReadDeadline(time.Now().Add(timeout))
+		c.nc.SetReadDeadline(c.clock.Now().Add(timeout))
```

Comment:
I recommend leaving this as `time.Now` because the deadline will be evaluated based on the real world clock. We'll have to do more work before we can simulate deadlines expiring.